iothub/device/tests/MessageTests.cs

```diff
@@ -165,5 +165,29 @@ public void SettingMessageAsSecurityMessageTest()
             Assert.AreEqual(CommonConstants.SecurityMessageInterfaceId, msg.SystemProperties[MessageSystemPropertyNames.InterfaceId]);
             Assert.IsTrue(msg.IsSecurityMessage);
         }
+
+        [TestMethod]
+        public void CloneWithBodyTest()
```

Comment:
Can we add a unit test around the scenario where Property/SystemProperty does have null value

Reply:
the check needs reflection to access the internal maps and see if the cloned instance has entries with explicit null-s where it was defined in the original, and missing entries where it was not defined in the original Messages.
Let me know if you are ok these sort of tests (not everybody is ok with testing private parts) or you have a suggestion for different approach

Reply:
You are right. The `Properies` property is private set, and initialized in the ctor with a dictionary. So it cannot be null.